src/jit/codegenarm.cpp

```diff
@@ -1230,6 +1230,56 @@ void CodeGen::genCodeForDivMod(GenTreeOp* tree)
 }
 
 //------------------------------------------------------------------------
+// genCkfinite: Generate code for ckfinite opcode.
+//
+// Arguments:
+//    treeNode - The GT_CKFINITE node
+//
+// Return Value:
+//    None.
+//
+// Assumptions:
+//    GT_CKFINITE node has reserved an internal register.
+//
+void CodeGen::genCkfinite(GenTreePtr treeNode)
+{
+    assert(treeNode->OperGet() == GT_CKFINITE);
+
+    emitter*   emit       = getEmitter();
+    GenTreePtr op1        = treeNode->gtOp.gtOp1;
+    var_types  targetType = treeNode->TypeGet();
+    regNumber  intReg     = treeNode->GetSingleTempReg();
+    regNumber  fpReg      = genConsumeReg(op1);
+
+    // Extract exponent into a register and check if it's all 1's
+    if (targetType == TYP_FLOAT)
+    {
+        emit->emitIns_R_R(INS_vmov_f2i, EA_4BYTE, intReg, fpReg);
+        emit->emitIns_R_R_I(INS_lsr, EA_4BYTE, intReg, intReg, 23);
+        emit->emitIns_R_R(INS_uxtb, EA_1BYTE, intReg, intReg);
+        emit->emitIns_R_I(INS_cmp, EA_4BYTE, intReg, 0xFF);
+    }
+    else // TYP_DOUBLE
+    {
+        emit->emitIns_R_R(INS_vmov_f2i, EA_4BYTE, intReg, REG_NEXT(fpReg));
+        emit->emitIns_R_R_I_I(INS_ubfx, EA_4BYTE, intReg, intReg, 20, 11);
+        emit->emitIns_R_I(INS_add, EA_4BYTE, intReg, 1);
+        emit->emitIns_R_I(INS_cmp, EA_4BYTE, intReg, 0x800);
```

Comment:
I guess we can get rid of `cmp` completely, if we replace `ubfx` by its signed variant `sbfx`. I'll check this hypothesis tomorrow, when I'll have access to my workplace.